src/shared/Core/HttpClientFactory.cs

```diff
@@ -301,5 +335,68 @@ public bool TryCreateProxy(out IWebProxy proxy)
             proxy = null;
             return false;
         }
+
+        public class CurlCookieParser
+        {
+            private readonly ITrace _trace;
+
+            public CurlCookieParser(ITrace trace)
+            {
+                _trace = trace;
+            }
+
+            public IList<Cookie> Parse(string content)
+            {
+                var cookies = new List<Cookie>();
+
+                // Parse the cookie file content
+                var lines = content.Split(new[] { '\r', '\n' }, StringSplitOptions.RemoveEmptyEntries);
+                for (int i = 3; i < lines.Length; i++)
+                {
+                    var parts = lines[i].Split(new[] { '\t' }, StringSplitOptions.RemoveEmptyEntries);
+                    if (parts.Length >= 7)
+                    {
+                        var domain = parts[0];
+                        domain = domain.StartsWith("#HttpOnly_") ? parts[0].Substring("#HttpOnly_".Length) : parts[0];
+                        domain = domain.TrimStart('.');
+                        var includeSubdomains = parts[1] == "TRUE";
+                        var path = parts[2];
+                        var secureOnly = parts[3] == "TRUE";
+                        var expires = parts[4];
+                        var name = parts[5];
+                        var value = parts[6];
+
+                        cookies.Add(new Cookie(name, value, path, domain)
+                        {
+                            Expires = ParseExpires(expires),
+                            Secure = secureOnly,
```

Comment:
This function doens't look correct to me. The epoch for `DateTime` is `0001-01-01T00:00:00Z`, **not** the UNIX time epoch of `1970-01-01T00:00:00Z`, and `DateTime.TryParse(string, out DateTime)` would, when given an integer, take this as a number of ticks since the .NET time epoch.

I would suggest an implementation similar to below, where we parse the string as a number of seconds since the UNIX epoch:

```csharp
            private static DateTime ParseExpires(string expires)
            {
#if NETFRAMEWORK
                DateTime epoch = new DateTime(1970, 01, 01, 0, 0, 0, DateTimeKind.Utc);
#else
                DateTime epoch = DateTime.UnixEpoch;
#endif

                if (!long.TryParse(expires, out long i))
                {
                    return epoch.AddSeconds(i);
                }

                return epoch;
			}
```

Sadly since we also target .NET Framework, and `DateTime.UnixEpoch` is [only defined as of .NET Core 2.1](https://learn.microsoft.com/en-us/dotnet/api/system.datetime.unixepoch?view=net-7.0) we need to manually create the epoch for .NET Framework.